packages/aws-cdk-lib/aws-eks/lib/cluster.ts

```diff
@@ -2681,3 +2731,92 @@ function cpuArchForInstanceType(instanceType: ec2.InstanceType) {
 function flatten<A>(xss: A[][]): A[] {
   return Array.prototype.concat.call([], ...xss);
 }
+
+function validateRemoteNetworkConfig(props: ClusterProps) {
+  // validate that no two CIDRs overlap within the same remote node network
+  props.remoteNodeNetworks!.forEach((network, index) => {
```

Comment:
I personally don't like the consistent non-null assertion usage of `!`. IMO, if we move the check `if (props.remoteNodeNetworks) {` in this function, we don't need the `!` right?

Something like
```
if (!props.remoteNodeNetworks) {
  return;
}
```

Reply:
That makes sense, pushed an update. Thanks!